### ISSUE TITLE
Feature Request: Support for cursor-based pagination in collections

### ISSUE THREAD

**Issue by metatronicx** [pocketbase/pocketbase#2680]
Currently, only offset-based pagination is what we can use when we want to retrieve data in separated small chunk to not overwhelm end users with a lot of data displayed on the uis. It would be great to also have cursor-based pagination and use created field as the default cursor. 

I've never touched go code... But I might be able to help with this if I have some guidance from you the original maintainers

**Comment by ganigeorgiev** [comment]
I think this is already supported but it depends on what field you want to use as cursor (_we don't have an autoincrement field at the moment_).

For example if you want to use the built-in `created`/`updated` date fields (dates have ms granularity), this could look something like:

```js
let lastCreated = "";

async function load() {
    const result = await pb.collection("example").getList(1, 30, {
        sort:   "created",
        filter: `created > "${lastCreated}"`,
    });

    if (result.items.length) {
        lastCreated = result.items[result.items.length - 1].created;
    }

    return result;
}

load();
```

**Comment by metatronicx** [comment]
Pretty straightforward. Thanks @ganigeorgiev ❤️

**Comment by joy-sec** [comment]
> I think this is already supported but it depends on what field you want to use as cursor (_we don't have an autoincrement field at the moment_).
> 
> For example if you want to use the built-in `created`/`updated` date fields (dates have ms granularity), this could look something like:
> 
> ```js
> let lastCreated = "";
> 
> async function load() {
>     const result = await pb.collection("example").getList(1, 30, {
>         sort:   "created",
>         filter: `created > "${lastCreated}"`,
>         skipTotal: true, // to skip the total count queries; available with v0.17+
>     });
> 
>     if (result.items.length) {
>         lastCreated = result.items[result.items.length - 1].created;
>     }
> 
>     return result;
> }
> 
> load();
> ```

There can be data inconsistency if multiple rows have the same createdAt value.

**Comment by ganigeorgiev** [comment]
@joy-sec As mentioned above it depends on the desired granularity. If date's milliseconds are not enough for your use case then you can define and maintain your own custom field.

In any case you are commenting to a very old discussion without really contributing anything to it. If you have an issue or other questions, please open a new Q&A discussion as the threaded format is more suitable for this type of posts.
